### ISSUE TITLE
Hide implementation details from try_scan!

### ISSUE THREAD

**Issue by kangalioo** [oli-obk/rust-si#30]
The following two macro variants seem to be more of an implementation detail than a part of the public API

```rust
(@question_mark: $($e:tt)+) => { ... };
(@unwrap: $($e:tt)+) => { ... };
```

When I first encountered it, I puzzled over them a good while to figure out what library users where supposed to achieve with this. Only after a while did I realize that this was an implementation detail for the bottom-most macro rule.

To avoid this confusion and clear up the docs, these two macro rules could be moved into a separate, private macro

**Comment by oli-obk** [comment]
Makes sense. Maybe we can also do something like `#[doc(hidden)]` or what that flag is? Not sure if that works on macro arms.
